src/NuGet.Clients/NuGet.CommandLine/Commands/InstallCommand.cs

```diff
@@ -350,18 +350,27 @@ private CommandLineSourceRepositoryProvider GetSourceRepositoryProvider()
                     PackageSaveMode = EffectivePackageSaveMode;
                 }
 
-                // Check if the package already exists or a higher version exists already.
-                var skipInstall = project.PackageExists(packageIdentity, PackageSaveMode);
+                var exactVersionExists = project.PackageExists(packageIdentity, PackageSaveMode);
+                var higherVersionExists = alreadyInstalledVersions.Any(e => e >= version);
 
-                // For SxS allow other versions to install. For non-SxS skip if a higher version exists.
-                skipInstall |= (ExcludeVersion && alreadyInstalledVersions.Any(e => e >= version));
+                // For SxS skip if exact version exists. For non-SxS skip if a higher version exists.
+                var skipInstall = exactVersionExists || (ExcludeVersion && higherVersionExists);
 
                 if (skipInstall)
                 {
-                    var message = string.Format(
-                        CultureInfo.CurrentCulture,
-                        LocalizedResourceManager.GetString("InstallCommandPackageAlreadyExists"),
-                        packageIdentity);
+                    string message;
+                    if (exactVersionExists) {
```

Comment:
We use [Allman style](http://en.wikipedia.org/wiki/Indent_style#Allman_style) braces, where each brace begins on a new line. One exception is that a using statement is permitted to be nested within another using statement by starting on the following line at the same indentation level, even if the nested using contains a controlled block.
```suggestion
                    if (exactVersionExists) 
                    {
```